File: plugins/node/template/addons/oauth/src/config/oauth.ts
import { z } from "zod";

const oauthSchema = z.object({
  GOOGLE_CLIENT_ID: z.string().min(1),
  GOOGLE_CLIENT_SECRET: z.string().min(1),
  GITHUB_CLIENT_ID: z.string().min(1),
  GITHUB_CLIENT_SECRET: z.string().min(1),
  APP_URL: z.string().url().default("http://localhost:3000"),
});

export const oauthEnv = oauthSchema.parse(process.env);

File: plugins/python/template/addons/oauth/app/core/oauth.py
from pydantic_settings import BaseSettings, SettingsConfigDict


class OAuthSettings(BaseSettings):
    GOOGLE_CLIENT_ID: str = ""
    GOOGLE_CLIENT_SECRET: str = ""
    GITHUB_CLIENT_ID: str = ""
    GITHUB_CLIENT_SECRET: str = ""
    APP_URL: str = "http://localhost:8000"

    model_config = SettingsConfigDict(env_file=".env", case_sensitive=True, extra="ignore")


oauth_settings = OAuthSettings()
